### DISCUSSION TITLE
About WAN configuration and example

### DISCUSSION THREAD

**Discussion by mengchaoheng** [eProsima/DDS-Router#348] [Q&A]
hello guy， you did a good job on DDS router. I want to test the use of dds router over 2 computer. I saw that the [case](https://eprosima-dds-router.readthedocs.io/en/latest/rst/examples/wan_example.html) in the tutorial needs to set my public ip and private ip, and I am confused about it. I checked my network settings, the private ip is fine, but the public ip is a "private IP"  provided by my ISP (Internet Service Provider), the real public ip resources are very limited, and I can't get it. This is very common in my area, almost all network devices go online this way. How should I set my ip to run the demo of wan communication in the tutorial?

**Comment by mengchaoheng** [comment]
@rsanchez15

**Comment by mengchaoheng** [comment]
If I use dds router in a private network built by myself, this network is not connected to the outside world, how should I set the public ip in this case？

**Comment by jparisu** [comment]
Hi @mengchaoheng .
We are glad to know you are using our product.

First of all, as this is a question and not an issue of the DDS Router, I will move it to Questions & Answers and I will answer you there.

**Comment by mengchaoheng** [comment]
Thank you very much, my usage scenario is very simple, it does not involve the connection of the internet, it is just the host connected to the LAN in the LAN of the other LAN, in order to explain my scenario I use a parallels virtual machine for illustration, his virtual machine network has [Bridged Network and Shared Networking](https://kb.parallels.com/4948) mode, when I set up two virtual machines working in these two modes to use dds router to communicate, how to set my IP? the parallels support setting port forwarding rules,and mac  have closed the  firewall.

one vm is Bridged Network

![image](https://user-images.githubusercontent.com/43166007/216811979-b94dff3e-2ec0-4741-b39e-e9c992c30fb5.png)

another vm is Shared Networking
![image](https://user-images.githubusercontent.com/43166007/216811959-f41ac7a5-5c8b-40cb-bf11-b6c222999ab4.png)

**Comment by mengchaoheng** [comment]
@jparisu

**Comment by jparisu** [comment]
About the problem with the public and private addresses that you are encountering:
The IP that you need to set in both client and server sides is the **Public IP**, that means, the IP that you ISP gives you to connect to the internet. You can find it easily searching in any navigator, there are some web pages that give this information (e.g. https://whatismyipaddress.com/ ). This IP is none that you show on your schemas, but the one from the router (192.168.0.1) to the Internet.
This IP is required because it is the only way for the 2 machines to know where to connect in order to communicate with each other. The protocol that uses DDS for discovery (multicast) is not available in the Internet.

In the other hand, if your problem is related with the instability of your Public IP (some ISP change Public IPs unannounced) I can give you some advises or ideas on how to solve it. But this is beyond the DDS Router scope, and the solutions depend on each specific scenario and each user:

1. Use DNS Domain names instead of IPs. Setting a custom DNS server in both machines can solve the problem, as you do not need to change the configuration every time the IP changes. This is done by using keyword `domain` instead of `ip` in yaml configuration.
2. Ask your ISP for a fixed IP. Most ISP support fixed IPs (normally by paying).
3. Use Cloud services. Most Cluod services use a fixed IP for a cluster. Check our Kubernetes example: https://eprosima-dds-router.readthedocs.io/en/latest/rst/use_cases/ros_cloud.html

**Reply by mengchaoheng** [reply]
Maybe you understand wrong, I don't need to use dds router in the internet, but only need to use it in a closed network environment, for example, two routers are connected in parallel under one router, and then two devices are connected to the two sub-routers respectively  , and finally this two devices communicate.

**Reply by mengchaoheng** [reply]
@jparisu

**Reply by mengchaoheng** [reply]
![image](https://user-images.githubusercontent.com/43166007/216814928-949826bf-9dfd-464b-93a6-281a59746bd3.jpeg)

**Reply by mengchaoheng** [reply]
@jparisu

**Comment by mengchaoheng** [comment]
I solved it, "Public IP" is just a relative concept. Thank you very much!
